### PR TITLE
Add dart2js support to the WebEntrypointBuilder

### PR DESCRIPTION
**Pull request by jakemac53**
See changelog for details.

Closes https://github.com/dart-lang/build/issues/617

### CODE REVIEW

File: build_runner/lib/src/generate/phase.dart [hunk 1]
```diff
@@ -94,14 +94,8 @@ class BuildAction implements InputMatcher {
...
   @override
-  int get hashCode => _deepEquals.hash([
-        '${builder.runtimeType}',
-        package,
-        _inputs,
-        isOptional,
-        hideOutput,
-        builderOptions.config
-      ]);
+  int get hashCode => _deepEquals.hash(
```
Comment (natebosch) [inline build_runner/lib/src/generate/phase.dart]: Can we send this out in a separate PR?

File: build_web_compilers/lib/src/web_entrypoint_builder.dart [hunk 2]
```diff
@@ -7,14 +7,64 @@ import 'dart:async';
...
+  DartDevc,
+}
+
+/// The top level keys supported for the `options` config for the
+/// [WebEntrypointBuilder].
+const _supportedOptions = const ['compiler', 'dart2js_args'];
+
+/// A builder which compiles entrypoints for the web.
+///
+/// Supports `dart2js` and `dartdevc` today.
```
Comment (natebosch) [inline build_web_compilers/lib/src/web_entrypoint_builder.dart]: [nit] 'today' shouldn't be necessary to specify (unless we have short term plans to change it)

File: build_web_compilers/test/dart2js_bootstrap_test.dart [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          print(hello);
+        }
+      ''',
+    };
+
+    // Set up all the other required inputs for this test.
+    await testBuilderAndCollectAssets(new ModuleBuilder(), assets);
+  });
+
+  test("can bootstrap dart entrypoints", () async {
```
Comment (natebosch) [inline build_web_compilers/test/dart2js_bootstrap_test.dart]: [nit] mixing quote types

Comment (jakemac53) [reply build_web_compilers/test/dart2js_bootstrap_test.dart]: done

File: build_web_compilers/lib/src/web_entrypoint_builder.dart (hunk 2 above)
Comment (jakemac53) [reply build_web_compilers/lib/src/web_entrypoint_builder.dart]: done

File: build_runner/lib/src/generate/phase.dart (hunk 1 above)
Comment (jakemac53) [reply build_runner/lib/src/generate/phase.dart]: sure
